ui/app/routes/application.js

```diff
@@ -25,7 +26,21 @@ export default class ApplicationRoute extends Route {
     }
   }
 
-  beforeModel(transition) {
+  async beforeModel(transition) {
+    let exchangeOneTimeToken;
+
+    if (transition.to.queryParams.ott) {
+      exchangeOneTimeToken = this.get('token.exchangeOneTimeToken')
+        .perform(transition.to.queryParams.ott)
+        .catch(() => {
+          this.controllerFor('application').set('error', new OTTExchangeError());
```

Comment:
I think ideally this error casting would happen in the data layer. Presumably in the token adapter.

Not a big deal though if that ends up being more trouble than it's worth.

Reply:
It was easy enough, makes sense, done in d5e2e5d, thank you!